### PR TITLE
feat: Show Staked ETH position in mobile homepage along with other tokens

### PR DESCRIPTION
**Pull request by nickewansmith**
## Explanation

**What is the current state of things and why does it need to change?**
- Currently the `metamask-mobile` app using the `assets-controllers` to get information about account token balances. We need to be able to support getting a new type of asset on mainnet and holesky chains, _Staked Ethereum_, which is not a token but represents the amount of ETH staked using our products.

**What is the solution your changes offer and how does it work?**
- We update the `AssetContractController` with a new method, `getStakedBalanceForChain`, which gets staked ethereum balances per account from the Stakewise vault contract. We update the AccountTrackerController with options to `includeStakedAssets` and to add a `getStakedBalanceForChain` method. 
- We bind `AssetContractController.getStakedBalanceForChain` to `getStakedBalanceForChain` option property in `metamask-mobile` code and then set `includeStakingAssets` option to the boolean feature flag for ETH Staking on Mobile. 
- We use the AccountTrackerController state in mobile to update the account `balance` and now, if enabled the `stakedBalance` as well.

**Are there any changes whose purpose might not obvious to those unfamiliar with the domain?**
- We don't want to show `stakedBalance` if not on a supported network, and so return undefined vs defaulting to zero hex. If there is an error and we are on a supported network, we want to default to zero hex.

**If your primary goal was to update one package but you found you had to update another one along the way, why did you do so?**
- There is 1 package affected

**If you had to upgrade a dependency, why did you do so?**
- No need to update dependency

## References

**Are there any issues that this pull request is tied to?**
- Relates to https://consensyssoftware.atlassian.net/browse/STAKE-817

**Are there other links that reviewers should consult to understand these changes better?**
- https://github.com/MetaMask/metamask-mobile/pull/12146 PR to MM mobile with patched `assets-controllers@38.3.0`

**Are there client or consumer pull requests to adopt any breaking changes?**
- No

### `@metamask/assets-controllers`

**ADDED**: AssetsContractController.getStakedBalanceForChain method to get staked ethereum balance for an address
**ADDED**: AccountTrackerController options `includeStakedEthereum` and `getStakedBalanceForChain` for turning on staked balance functionality and providing a method to do so

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate
- [x] I've prepared draft pull requests for clients and consumer packages to resolve any breaking changes

### CODE REVIEW

File: packages/assets-controllers/src/AssetsContractController.ts [hunk 1]
```diff
@@ -688,6 +702,78 @@ export class AssetsContractController {
...
+      },
+    ];
+
+    try {
+      const contract = new Contract(contractAddress, abi, provider);
+      const exchangeRateDenominator = BigNumber.from((1e18).toString());
+      const multiplier = exchangeRateDenominator;
+      const userShares = await contract.getShares(address);
+
+      // convert shares to assets only if address shares > 0 else return default balance
```
Comment (amitabh94) [inline packages/assets-controllers/src/AssetsContractController.ts:757]: Do we need to calculate exchange rate here or can we directly get assets from convertToAssets function ?

**Review by amitabh94 — COMMENTED** [review]
Just some questions but overall LGTM!

File: packages/assets-controllers/src/AccountTrackerController.test.ts [hunk 2]
```diff
@@ -248,6 +248,123 @@ describe('AccountTrackerController', () => {
...
+                    balance: '0x0',
+                  },
+                },
+              },
+            });
+          },
+        );
+      });
+
+      it('should update staked balance when includeStakedAssets and multi-account is enabled', async () => {
```
[the remark this answers is not in the document]
Comment (nickewansmith) [reply packages/assets-controllers/src/AccountTrackerController.test.ts:329]: This feature is not currently enabled in production but allows all accounts to get balances refreshed and not just the currently selected account. I added a test here to make sure that the staked balance feature is compatible.

File: packages/assets-controllers/src/AssetsContractController.ts (hunk 1 above)
Comment (nickewansmith) [reply packages/assets-controllers/src/AssetsContractController.ts:757]: Hm, you're right we don't really need to get the rate here. Thx will make that change!

**Review by amitabh94 — APPROVED** [review]
LGTM!

**Pull request merged**